src/lightning_app/cli/lightning_cli_create.py

```diff
@@ -0,0 +1,86 @@
+import click
+
+from lightning_app.cli.cmd_clusters import _check_cluster_name_is_valid, _default_instance_types, AWSClusterManager
+
+
+@click.group("create")
+def create():
+    """Create Lightning AI BYOC managed resources."""
+    pass
+
+
+@create.command("cluster")
+@click.argument("cluster_name", callback=_check_cluster_name_is_valid)
+@click.option("--provider", "provider", type=str, default="aws", help="cloud provider to be used for your cluster")
+@click.option("--external-id", "external_id", type=str, required=True)
+@click.option(
+    "--role-arn", "role_arn", type=str, required=True, help="AWS role ARN attached to the associated resources."
+)
+@click.option(
+    "--region",
+    "region",
+    type=str,
+    required=False,
+    default="us-east-1",
+    help="AWS region that is used to host the associated resources.",
+)
+@click.option(
+    "--instance-types",
+    "instance_types",
+    type=str,
+    required=False,
+    default=",".join(_default_instance_types),
+    help="Instance types that you want to support, for computer jobs within the cluster.",
+)
+@click.option(
+    "--cost-savings",
+    "cost_savings",
+    type=bool,
+    required=False,
+    default=False,
+    is_flag=True,
+    help=""""Use this flag to ensure that the cluster is created with a profile that is optimized for cost savings.
+        This makes runs cheaper but start-up times may increase.""",
+)
+@click.option(
+    "--edit-before-creation",
+    default=False,
+    is_flag=True,
+    help="Edit the cluster specs before submitting them to the API server.",
+)
+@click.option(
+    "--wait",
+    "wait",
+    type=bool,
+    required=False,
+    default=False,
+    is_flag=True,
+    help="Enabling this flag makes the CLI wait until the cluster is running.",
+)
+def create_cluster(
```

Comment:
Have you been considering using [Fire](https://github.com/google/python-fire)? That would have less boilerplate...

Reply:
In that case, shouldn't we migrate the entire CLI from click to fire?

Reply:
I would be in favor, but let's chat about it in another issue :)
May I ask you to open an issue here to check feasibility and eventual advantages :rabbit:

Reply:
for sure, I've posted https://github.com/Lightning-AI/lightning/issues/13931